blacklight-cornell/lib/blacklight_cornell/cornell_catalog.rb

```diff
@@ -536,26 +549,23 @@ def cjk_mm_val
     silence_warnings { @@cjk_mm_val = '3<86%'}
   end
 
+  # Update in blacklight_range_limit v.9 
   def check_dates(params)
+    alert = nil
     pub_date_facet = params[:range][:pub_date_facet]
     unknown_pub_date_facet = params[:range]['-pub_date_facet']
-    # check for missing or unknown Publication Year
+    begin_year = Integer(pub_date_facet[:begin]) rescue nil
+    end_year = Integer(pub_date_facet[:end]) rescue nil
     if unknown_pub_date_facet.present? || pub_date_facet[:missing].present? || (pub_date_facet[:begin].blank? && pub_date_facet[:end].blank?)
-      return
-    end
-    # crashes later on if begin > end so raise exception here
-    begin_test = Integer(params[:range][:pub_date_facet][:begin]) rescue nil
-    end_test = Integer(params[:range][:pub_date_facet][:end]) rescue nil
-    min_year = 0
-    unless begin_test.present? && begin_test >= min_year
-      raise ArgumentError.new(I18n.t('blacklight.search.errors.publication_year_range.begin'))
-    end
-    unless end_test.present? && end_test >= min_year
-      raise ArgumentError.new(I18n.t('blacklight.search.errors.publication_year_range.end'))
-    end
-    unless begin_test <= end_test
-      raise ArgumentError.new(I18n.t('blacklight.search.errors.publication_year_range.order'))
-    end
+      alert
+    elsif begin_year.blank? || begin_year.negative?
```

Comment:
Technically, I think blacklight_range_limit supports negative numbers. If you do: /?utf8=✓&advanced_query=yes&boolean_row%5B1%5D=AND&op_row%5B%5D=AND&op_row%5B%5D=AND&q=&q_row%5B%5D=cats+dogs+gerbils&q_row%5B%5D=&search_field=advanced&search_field_row%5B%5D=all_fields&search_field_row%5B%5D=all_fields&sort=score+desc%2C+pub_date_sort+desc%2C+title_sort+asc&range%5Bpub_date_facet%5D%5Bbegin%5D=-5&range%5Bpub_date_facet%5D%5Bend%5D=2019, that search seems to work, but it displays a flash message. Should we remove the negative checks?

Reply:
I debated this as I worked on this issue. You are correct that it will return a range that includes a negative number, but the publication range graph seems to imply that the assumed range does not actually go below zero. 

<img width="412" height="387" alt="Screenshot 2025-08-22 at 12 03 57 PM" src="https://github.com/user-attachments/assets/817799cd-6c47-4e11-b77d-b437f841867e" />

We also don't seem to have any records with dates that go into the negatives (see /?q=&range%5Bpub_date_facet%5D%5Bbegin%5D=-2000&range%5Bpub_date_facet%5D%5Bend%5D=-1&search_field=all_fields), so it's hard to know whether or not it would return records if we did have any.

You're probably right, though, that we don't need the flash message.